common/scala/src/main/scala/whisk/core/mesos/MesosTask.scala

```diff
@@ -0,0 +1,163 @@
+    val mesosRam = memory.toMB.toInt
+
+    //TODO: update mesos-actor to support multiple param values for the same key via Map[String, Set[String]]
+    val flatParams = parameters.filter(_._2.nonEmpty).map(e => (e._1 -> e._2.head))
+    val task = new TaskDef(
+      taskId,
+      name.getOrElse(image), //task name either the indicated name, or else the image name
+      image,
+      mesosCpuShares,
+      mesosRam,
+      List(8080), //all action containers listen on 8080
+      Some(0), //port at index 0 used for health
+      false,
+      Bridge,
+      flatParams,
+      environment)
+
+    val launched: Future[Running] =
+      mesosClientActor.ask(SubmitTask(task))(taskLaunchTimeout).mapTo[Running]
+
+    launched.map(taskDetails => {
+      val taskHost = taskDetails.hostname
+      val taskPort = taskDetails.hostports(0)
+      log.info(this, s"launched task with state ${taskDetails.taskStatus.getState} at ${taskHost}:${taskPort}")
+      val containerIp = new ContainerAddress(taskHost, taskPort)
+      val containerId = new ContainerId(taskId);
+      new MesosTask(containerId, containerIp, ec, log, taskId, mesosClientActor, mesosUri)
+    }) recover {
+      case t => throw new Exception(t)
+    }
+
+  }
+
+}
+
+object JsonFormatters extends DefaultJsonProtocol {
+  implicit val createContainerJson = jsonFormat3(CreateContainer)
+}
+class MesosTask(override protected val id: ContainerId,
+                override protected val addr: ContainerAddress,
+                override protected val ec: ExecutionContext,
+                override protected val logging: Logging,
+                taskId: String,
+                mesosClientActor: ActorRef,
+                mesosUri: String)
+    extends Container {
+
+  implicit val e = ec
+
+  /** Stops the container from consuming CPU cycles. */
+  override def suspend()(implicit transid: TransactionId): Future[Unit] = {
+    //suspend not supported
+    Future(Unit)
+  }
+
+  /** Dual of halt. */
+  override def resume()(implicit transid: TransactionId): Future[Unit] = {
+    //resume not supported
+    Future(Unit)
+  }
+
+  /** Completely destroys this instance of the container. */
+  override def destroy()(implicit transid: TransactionId): Future[Unit] = {
+    mesosClientActor
+      .ask(DeleteTask(taskId))(MesosTask.taskDeleteTimeout)
+      .mapTo[TaskStatus]
+      .map(taskStatus => {
+        logging.info(this, s"task killed ended with state ${taskStatus.getState}")
+      }) recover {
+      case t => throw new Exception(t)
+    }
+  }
+
+  /** Obtains logs up to a given threshold from the container. Optionally waits for a sentinel to appear. */
+  override def logs(limit: ByteSize, waitForSentinel: Boolean)(
+    implicit transid: TransactionId): Future[Vector[String]] =
+    Future.successful(
+      Vector(
+        "Logs are not collected from Mesos containers currently.",
+        s"You can browse the logs for Mesos Task ID ${taskId} using the mesos UI at ${mesosUri}"))
```

Comment:
This is not specific to a subject/namespace, hence logs are visible to all users of the system in this model? With the log draining you’re imagining this will change so that two subjects’ logs are visible to each other?

Reply:
Yes, this is a stopgap until LogStore is available; Currently mesos integration provides no isolation of two subjects logs. In general devs should not even have access to the mesos UI, (or the logs of other subjects' actions).